mender-update/cli/actions.cpp

```diff
@@ -28,9 +39,66 @@ namespace conf = mender::common::conf;
 namespace database = mender::common::key_value_database;
 namespace events = mender::common::events;
 namespace log = mender::common::log;
-
-namespace daemon = mender::update::daemon;
+namespace error = mender::common::error;
+namespace kv_db = mender::common::key_value_database;
+namespace path = mender::common::path;
 namespace standalone = mender::update::standalone;
+namespace daemon = mender::update::daemon;
+
+
+error::Error maybeInstallBootstrapArtifact(context::MenderContext &main_context) {
+	const string bootstrap_artifact_path {
+		main_context.GetConfig().paths.GetBootstrapArtifactFile()};
+	// Check if the DB is populated - then install conditionally
+	auto &db = main_context.GetMenderStoreDB();
+	auto exp_key = db.Read(main_context.artifact_name_key);
+	if (exp_key) {
+		// Key exists. Do nothing
+		return error::NoError;
+	}
+	error::Error err = exp_key.error();
+	if (err != error::NoError
```

Comment:
`error::NoError` should never be used with `expected::unexpected()`, right? We should have an `assert()` or check there.

Reply:
Hmm, what do you mean?

We do this all the time? 

Remember that the case is inverted here, we check for `success` above. This is the `error` condition.

Reply:
> They should never contain error::NoError, AFAICT. Hence that part of the condition is unnecessary/confusing.

And actually, the test for `KeyError` implies that `err != error::NoError`, so it's unnecessary.

Reply:
Ahh, now I see what you mean